### DISCUSSION TITLE
Is it possible to set a custom domain

### DISCUSSION THREAD

**Discussion by KristaNEO** [ScarletKuro/Blazor.GoogleTagManager#1] [Q&A] [answered]
Like in this example

https://developers.google.com/tag-platform/tag-manager/server-side/custom-domain?option=same-origin

**Comment by ScarletKuro** [comment, marked as answer]
What I believe you could do is to copy paste the script to `wwwroot`:
https://github.com/ScarletKuro/Blazor.GoogleTagManager/blob/main/src/Blazor.GoogleTagManager/wwwroot/GoogleTagManager.js

Change the domain on your own one, and during setup disable `ImportJsAutomatically` and import your own script manually:
```
<script src="_content/{you_assembly_name}/GoogleTagManager.js"></script>
```
don't forget to replace the `you_assembly_name` with your project name.

**Reply by KristaNEO** [reply]
Thank you. This seems like it could work, however I'm running into some sort of timing error with the scripts


Error: Microsoft.JSInterop.JSException: Could not find 'initialize' ('initialize' was undefined).
Error: Could not find 'initialize' ('initialize' was undefined).

**Reply by ScarletKuro** [reply]
did you remove the `﻿export` from js functions when copy pasting [GoogleTagManager.js](https://github.com/ScarletKuro/Blazor.GoogleTagManager/blob/main/src/Blazor.GoogleTagManager/wwwroot/GoogleTagManager.js)?
I forgot to mention it. The `export` is only when you use import, but it doesn't work with the script tag.

**Comment by KristaNEO** [comment]
Brilliant! Thank you that was it.

**Reply by ScarletKuro** [reply]
Glad it worked. Maybe next time if I will do a breaking change I will add a configurable url in options,

**Reply by KristaNEO** [reply]
Really like that idea, for now I'm sending the url as an attribute and customized the js which seems to work but will have to maintain as version updates are rolled out.
